File: tools/vibeue_call.mjs
import { readFile } from "node:fs/promises";
const BASE_URL = "http://127.0.0.1:8088/mcp";
const AUTH_TOKEN = JSON.parse(await readFile(new URL("../.mcp.json", import.meta.url), "utf8")).mcpServers.VibeUE.headers.Authorization;
const PROTOCOL_VERSION = "2025-11-25";

async function post(payload, sessionId = null) {
  const headers = {
    Authorization: AUTH_TOKEN,
    "Content-Type": "application/json",
    Accept: "application/json",
    "MCP-Protocol-Version": PROTOCOL_VERSION,
    Connection: "close",
  };
  if (sessionId) headers["Mcp-Session-Id"] = sessionId;

  const res = await fetch(BASE_URL, {
    method: "POST",
    headers,
    body: JSON.stringify(payload),
  });
  const text = await res.text();
  return {
    sessionId: res.headers.get("mcp-session-id"),
    json: text.trim() ? JSON.parse(text) : null,
  };
}

async function main() {
  if (process.argv.length < 4) {
    console.error("Usage: node vibeue_call.mjs <tool_name> <json_args|@file>");
    process.exit(2);
  }

  const toolName = process.argv[2];
  let rawArgs = process.argv[3];
  if (rawArgs.startsWith("@")) {
    rawArgs = await (await import("node:fs/promises")).readFile(rawArgs.slice(1), "utf8");
  }
  const toolArgs = JSON.parse(rawArgs);

  const init = await post({
    jsonrpc: "2.0",
    id: 1,
    method: "initialize",
    params: {
      protocolVersion: PROTOCOL_VERSION,
      capabilities: {},
      clientInfo: { name: "codex-local-client", version: "1.0" },
    },
  });
  const sessionId = init.sessionId;

  await post(
    { jsonrpc: "2.0", method: "notifications/initialized", params: {} },
    sessionId,
  );

  const result = await post(
    {
      jsonrpc: "2.0",
      id: 2,
      method: "tools/call",
      params: { name: toolName, arguments: toolArgs },
    },
    sessionId,
  );

  console.log(JSON.stringify(result.json, null, 2));
}

main().catch((error) => {
  console.error(error);
  process.exit(1);
});

File: tools/vibeue_call.py
import json
import sys
import urllib.request
from pathlib import Path

BASE_URL = "http://127.0.0.1:8088/mcp"
AUTH_TOKEN = json.loads((Path(__file__).resolve().parents[1] / ".mcp.json").read_text(encoding="utf-8"))["mcpServers"]["VibeUE"]["headers"]["Authorization"]
PROTOCOL_VERSION = "2025-11-25"


def post(payload, session_id=None):
    body = json.dumps(payload).encode("utf-8")
    headers = {
        "Authorization": AUTH_TOKEN,
        "Content-Type": "application/json",
        "Accept": "application/json",
        "MCP-Protocol-Version": PROTOCOL_VERSION,
    }
    if session_id:
        headers["Mcp-Session-Id"] = session_id
    req = urllib.request.Request(BASE_URL, data=body, headers=headers, method="POST")
    with urllib.request.urlopen(req, timeout=60) as resp:
        text = resp.read().decode("utf-8")
        session = resp.headers.get("mcp-session-id")
    if not text.strip():
        return session, None
    return session, json.loads(text)


def main():
    if len(sys.argv) < 3:
        print("Usage: vibeue_call.py <tool_name> <json_args|@file|- >", file=sys.stderr)
        sys.exit(2)

    tool_name = sys.argv[1]
    raw_args = sys.argv[2]
    if raw_args == "-":
        raw_args = sys.stdin.read()
    elif raw_args.startswith("@"):
        with open(raw_args[1:], "r", encoding="utf-8") as fh:
            raw_args = fh.read()
    tool_args = json.loads(raw_args)

    session_id, _ = post(
        {
            "jsonrpc": "2.0",
            "id": 1,
            "method": "initialize",
            "params": {
                "protocolVersion": PROTOCOL_VERSION,
                "capabilities": {},
                "clientInfo": {"name": "codex-local-client", "version": "1.0"},
            },
        }
    )
    post(
        {"jsonrpc": "2.0", "method": "notifications/initialized", "params": {}},
        session_id=session_id,
    )
    _, result = post(
        {
            "jsonrpc": "2.0",
            "id": 2,
            "method": "tools/call",
            "params": {"name": tool_name, "arguments": tool_args},
        },
        session_id=session_id,
    )
    print(json.dumps(result, ensure_ascii=False, indent=2))


if __name__ == "__main__":
    main()
